pkg/gardenlet/controller/shoot/shoot_care_control.go

```diff
@@ -134,7 +134,17 @@ func (c *defaultCareControl) Care(shootObj *gardencorev1beta1.Shoot, key string)
 
 	shootLogger.Debugf("[SHOOT CARE] %s", key)
 
-	operation, err := operation.New(shoot, c.config, shootLogger, c.k8sGardenClient, c.k8sGardenCoreInformers, c.identity, c.secrets, c.imageVector)
+	operation, err := operation.
+		NewBuilder().
+		WithLogger(shootLogger).
+		WithConfig(c.config).
+		WithGardenerInfo(c.identity).
+		WithSecrets(c.secrets).
+		WithImageVector(c.imageVector).
+		WithGardenFrom(c.k8sGardenCoreInformers, shoot.Namespace).
+		WithSeedFrom(c.k8sGardenCoreInformers, *shoot.Spec.SeedName).
+		WithShootFrom(c.k8sGardenCoreInformers, shoot).
```

Comment:
These 3 methods are only used here and in the function they add to the builder, we already have a garden client (see line 147). So I think, we should not need to pass the informer here.
Also, in general it would probably be more convenient to pass only the names to the builder methods and pass the garden client once to `Build()` and propagate it to the respective builder funcs there. @rfranzke @timuthy WDYT?

```suggestion
		WithGardenAndShootNamespace(shoot.Namespace).
		WithSeedName(*shoot.Spec.SeedName).
		WithShootName(shoot).
```

Reply:
obsolete with https://github.com/gardener/gardener/pull/2217#discussion_r417100845